src/pytorch_lightning/loggers/wandb.py

```diff
@@ -300,7 +301,7 @@ def __init__(
             name=name,
             project=project,
             id=version or id,
-            dir=save_dir,
+            dir=save_dir or kwargs.pop("dir"),
```

Comment:
Make sure `save_dir` is not overridden by `None` `dir`, see https://github.com/Lightning-AI/lightning/issues/14225.